### ISSUE TITLE
Only 5 seconds of video uploaded

### ISSUE THREAD

**Issue by brylie** [bibanon/tubeup#74]
The tubeup script seems to complete successfully on a 45 minute video. However, when I view the video on the Internet Archive, it is only 5 seconds in length:

Original video: https://www.youtube.com/watch?v=kDbYCsvGn4s
IA upload: https://archive.org/details/youtube-kDbYCsvGn4s

**Comment by brylie** [comment]
It appears to be the *last* five seconds of the video.

**Comment by brandongalbraith** [comment]
Is the item still in a derive state?

**Comment by brylie** [comment]
I tried just using `youtube-dl`, and the downloaded file is the last five seconds. Are you referring to YouTube 'derive state' or Internet Archive?

**Comment by brylie** [comment]
In any case, the YouTube video is not in 'Processing' state and Internet Archive video is not in 'derive' state. I think this is related to `youtube-dl`, so will open an issue there. Thank you for your prompt response @brandongalbraith :-)

**Comment by brandongalbraith** [comment]
Happy to help! Feel free to tag me on the youtube-dl issue.

**Comment by vxbinaca** [comment]
Have you re-produced this with youtube-dl?

**Comment by brylie** [comment]
Yes, with the command `youtube-dl https://www.youtube.com/watch?v=kDbYCsvGn4s`

Now that I have retried it, I am noticing that there may be some background process after the `youtube-dl` step where the video is transcoding. Perhaps tubeup is uploading the video before local transcoding is complete.

**Comment by brylie** [comment]
In effect, I see that youtube-dl downloads the videos, but there is only one very small MP4 file directly after the download. Then, after a couple of minutes, the MP4 is much larger, like it was transcoded.

**Comment by brylie** [comment]
Nevermind alltogether. It must be something that is going on with YouTube in the background. Now that I try to reproduce, things work as expected.

**Comment by vxbinaca** [comment]
He's referring to the item on IA deriving but youtube also derives just much much faster Wait an hour before mirroring your videos.. 

Now, I've run into this too and it's not a problem on our end but here's how you fix this:

`ia delete youtube-kDbYCsvGn4s ; tubeup kDbYCsvGn4s'

Then delete the items `history` folder. That will fix this.

Is there a fix for Tubeup for this? No, because it's a youtube problem. Also a Twitch problem.
Is there a youtube-dl fix for this? No, because the service has to process a multiple gigabyte file.

There is no prevention for this, just fixes on your end manually. Closing.

**Comment by brylie** [comment]
Thanks so much for your help @vxbinaca! :-)

**Comment by vxbinaca** [comment]
tl;dr: You're ripping it before youtube processed the stream fully. Just slow down give it sometime.
